domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt

```diff
@@ -0,0 +1,278 @@
+package org.oppia.domain.topic
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.model.ChapterSummary
+import org.oppia.app.model.ChapterSummary.Playability
+import org.oppia.app.model.LessonThumbnailGraphic
+import org.oppia.app.model.SkillSummary
+import org.oppia.app.model.StorySummary
+import org.oppia.app.model.Topic
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [TopicController]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class TopicControllerTest {
+  @Inject
+  lateinit var topicController: TopicController
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testRetrieveTopic_validTopic_isSuccessful() {
+    val topicLiveData = topicController.getTopic(TEST_TOPIC_ID_0)
+
+    val topicResult = topicLiveData.value
+    assertThat(topicResult).isNotNull()
+    assertThat(topicResult!!.isSuccess()).isTrue()
+  }
+
+  @Test
+  fun testRetrieveTopic_validTopic_returnsCorrectTopic() {
+    val topicLiveData = topicController.getTopic(TEST_TOPIC_ID_0)
+
+    val topic = topicLiveData.value!!.getOrThrow()
+    assertThat(topic.topicId).isEqualTo(TEST_TOPIC_ID_0)
+  }
+
+  @Test
+  fun testRetrieveTopic_validTopic_returnsTopicWithDescription() {
+    val topicLiveData = topicController.getTopic(TEST_TOPIC_ID_0)
+
+    val topic = topicLiveData.value!!.getOrThrow()
+    assertThat(topic.description).isEqualTo("A topic investigating the interesting aspects of the Oppia Android app.")
+  }
+
+  @Test
+  fun testRetrieveTopic_validTopic_returnsTopicWithStories() {
+    val topicLiveData = topicController.getTopic(TEST_TOPIC_ID_0)
+
+    val topic = topicLiveData.value!!.getOrThrow()
+    assertThat(getStoryIds(topic)).containsExactly(TEST_STORY_ID_0, TEST_STORY_ID_1).inOrder()
+  }
+
+  @Test
+  fun testRetrieveTopic_validTopic_returnsTopicWithSkills() {
+    val topicLiveData = topicController.getTopic(TEST_TOPIC_ID_0)
+
+    val topic = topicLiveData.value!!.getOrThrow()
+    assertThat(getSkillIds(topic)).containsExactly("test_skill_id_0", "test_skill_id_1").inOrder()
+  }
+
+  // This test is intentionally verifying that there are no thumbnails for skills, since there are not yet any to
```

Comment:
Are we supposed to do multiline comments like this ?

Reply:
Multi-line comments can be done like this, or as a block comment, e.g.:

```kotlin
/*
 * First line
 * Second line.
 */
```

Or more compactly:

```kotlin
/* First line.
    Second line. */
```

Note the single '*' in the beginning to indicate this isn't a Javadoc.

In this case, I'm just providing context on why this test is set up in this way. This could be framed as either single-line comments or a documentation comment, though your point makes me think it might be better represented as documentation. Updated.